tests/conftest.py

```diff
@@ -87,6 +87,41 @@ def wait():
     )
 
 
+@pytest.fixture
+def set_session_cookie(selenium, base_url, request):
+    """Fixture used when we want to open an AMO page with a sessionid
+    cookie (i.e. a logged-in user) already set"""
+    # the user file is passed in the test as a marker argument
+    marker = request.node.get_closest_marker('user_data')
+    user_file = marker.args[0]
+    with open(f'{user_file}.txt', 'r') as file:
+        cookie_value = str(file.read())
+    # need to set the url context if we want to apply a cookie
+    # in order to avoid InvalidCookieDomainException error
+    selenium.get(base_url)
+    # set the sessionid cookie
+    create_session = selenium.add_cookie(
+        {
+            "name": "sessionid",
+            "value": cookie_value,
+        }
+    )
+    return create_session
+
+
+@pytest.fixture
+def destroy_file(request):
+    """Fixture to delete user files created for a test suite"""
+    marker = request.node.get_closest_marker('user_data')
+    user_file = marker.args[0]
+    import os
+
+    if os.path.exists(f'{user_file}.txt'):
+        os.remove(f'{user_file}.txt')
+    else:
+        print("The file does not exist")
+
+
```

Comment:
Why do we need to store a cookie if we are going to immediately use it. Will the login data not be stored during the test run, or are you trying to limit the amount of logins.

Reply:
@jrbenny35 for a couple of reasons:
1. to reduce the number of logins, as you mentioned - this will speed up test runs significantly if we're goin to use it for the other tests that we already have
2. to be able to write API tests that need authentication (which is done with the help of the session cookie); currently, there is no other method/infrastructure that would allow me to fetch the cookie other than logging in on AMO through the browser
3. can serve as a workaround for the 2fa deadlock we talked bout

Reply:
> can serve as a workaround for the 2fa deadlock we talked about

Can you elaborate on this?

Reply:
Okay great! I understand now. I think we should just use 1 selenium fixture and turn it into a factory based on a marker. So say we have a ```@pytest.mark.log_in``` then the selenium fixture should login and return a selenium instance with the correct session cookies set and the login done. This fixture can also use the ```tmpdir``` fixture to store the session cookie without having to manage removing it.

Reply:
@jrbenny35 I've used your suggestion and broadened the scope of the selenium fixture, but I had to base it on more than one marker, for the following reasons:
- `@pytest.mark.login` - this marker handles the normal browser login and stores the session cookie;  
- `@pytest.mark.create_session` - this is the marker that starts the browser with an active session (applies the cookie stored from  `@pytest.mark.login`  in order to avoid fxa logins completely)
- `@pytest.mark.clear_session` - this marker handles the session invalidation (logs out the user after the test suit is done) and deletes the session file

I've also looked into the `tmpdir` fixture, I've tried to use it to see how it works, but I wasn't very happy about the way it handles files, at least for my purpose. More specific, the files created by `tmpdir` are rewritten whenever a new test session starts, but they are not automatically deleted when a session ends, which means that the data stored in the file will continue to exist until a new session is started and overwrites it  - and so on and so forth. 
In my case, I want to make sure that any file created during a test session is completely removed when the session ends, hence I kept using my previous method for creating and deleting files, but now everything is incorporated in the selenium fixture.